### PR TITLE
Add Public Dataset Feature

### PR DESCRIPTION
**Pull request by harsha-accenture**
Added the functionality to show public datasets in the left panel.

### CODE REVIEW

File: src/dpms/bigQueryWidget.tsx [hunk 1]
```diff
@@ -323,83 +370,91 @@ const BigQueryComponent = ({
...
-              <div role="img" className="table-icon" onClick={handleIconClick}>
-                <iconTable.react
-                  tag="div"
-                  className="icon-white logo-alignment-style"
-                />
-              </div>
-            </>
-          );
-        }
+      // if (searchTerm) {
```
Comment (ptwng) [inline src/dpms/bigQueryWidget.tsx:373]: Did we intend to comment all of this out?

**Review by ojarjur — CHANGES_REQUESTED** [review]
Please delete the `Utilities.py` file and instead use the `gcp_project` method from the `google.cloud.jupyter_config.config` package to get the configured project.

File: dataproc_jupyter_plugin/contollers/bigqueryController.py [hunk 2]
```diff
@@ -25,17 +25,19 @@
     BigQueryTableInfoService,
     BigQueryTableListService,
 )
-
+from dataproc_jupyter_plugin.utils.utilities import Utilities
```
Comment (ojarjur) [inline dataproc_jupyter_plugin/contollers/bigqueryController.py:28]: Remove this `Utilities` package and instead import

```py
from google.cloud.jupyter_config import gcp_project
```

File: dataproc_jupyter_plugin/contollers/bigqueryController.py [hunk 3]
```diff
@@ -115,10 +121,10 @@ class BigqueryProjectsController(APIHandler):
     @tornado.web.authenticated
     def get(self):
         try:
-            bq_preview = BigQueryProjectService()
-            credentials = handlers.get_cached_credentials(self.log)
-            preview_data = bq_preview.bigquery_preview_data(credentials, self.log)
-            self.finish(json.dumps(preview_data))
+            project_list_cmd = "gcloud config config-helper --format=json --flatten=configuration.properties.core.project"
```
Comment (ojarjur) [inline dataproc_jupyter_plugin/contollers/bigqueryController.py:124]: This entire block of code can be replaced with:

```py
            project_list = [gcp_project(), bq_public_dataset_project_id]
```

File: src/dpms/bigQueryWidget.tsx (hunk 1 above)
Comment (harsha-accenture) [reply src/dpms/bigQueryWidget.tsx]: As we did not have the api, we had commented the code. Now it is all implemented.

File: dataproc_jupyter_plugin/contollers/bigqueryController.py (hunk 2 above)
Comment (harsha-accenture) [reply dataproc_jupyter_plugin/contollers/bigqueryController.py:29]: Changes implemented

File: dataproc_jupyter_plugin/utils/utilities.py [hunk 4]
```diff
@@ -0,0 +1,26 @@
+# Copyright 2024 Google LLC
```
Comment (ojarjur) [inline dataproc_jupyter_plugin/utils/utilities.py:1]: Please delete this file. This is both a terrible name for a package, and it reimplements functionality we already have from the `google.cloud.jupyter_config` package.

File: dataproc_jupyter_plugin/services/bigqueryService.py [hunk 5]
```diff
@@ -160,6 +155,56 @@ def bigquery_preview_data(self, credentials, dataset_id, table_id, page_token, l
...
+                access_token = credentials["access_token"]
+                project_id = credentials['project_id']
+                api_endpoint = (
+                    f"{datacatalog_url}v1/catalog:search"
+                )
+                headers = {
+                "Content-Type": CONTENT_TYPE,
+                "Authorization": f"Bearer {access_token}",
+                'X-Goog-User-Project' : project_id
+                }
```
Comment (ojarjur) [inline dataproc_jupyter_plugin/services/bigqueryService.py:175]: This formatting looks wrong. Did you remember to run the `black` command line tool over the modified files?

File: dataproc_jupyter_plugin/contollers/bigqueryController.py (hunk 2 above)
Comment (ojarjur) [reply dataproc_jupyter_plugin/contollers/bigqueryController.py:29]: This is not done yet. You still need to delete this import and remove the `utilities.py` file from the repository.

Comment (harsha-accenture) [reply dataproc_jupyter_plugin/contollers/bigqueryController.py]: Now deleted.

File: dataproc_jupyter_plugin/utils/utilities.py (hunk 4 above)
Comment (harsha-accenture) [reply dataproc_jupyter_plugin/utils/utilities.py]: Deleted

**Review by ojarjur — APPROVED** [review]
Approval on the Python changes. I still want Pat to review the Typescript changes.

**Review by ptwng — APPROVED** [review]

**Pull request merged**